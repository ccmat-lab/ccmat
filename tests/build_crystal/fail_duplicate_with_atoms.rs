use commat::{Atom, CrystalBuilder, Lattice};

fn main() {
    let lattice = Lattice::default();
    let atoms = vec![Atom::new([0.0, 0.0, 0.0], 8)];
    let _ = CrystalBuilder::new()
        .with_lattice(lattice)
        .with_atoms(&atoms)
        .with_atoms(&atoms)
        .build()
        .unwrap();
}
