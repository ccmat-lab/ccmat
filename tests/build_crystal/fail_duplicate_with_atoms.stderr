error[E0599]: no method named `with_atoms` found for struct `CrystalBuilder<LatticeSet, AtomsSet>` in the current scope
  --> tests/build_crystal/fail_duplicate_with_atoms.rs:13:10
   |
10 |       let _ = CrystalBuilder::new()
   |  _____________-
11 | |         .with_lattice(lattice)
12 | |         .with_atoms(&atoms)
13 | |         .with_atoms(&atoms)
   | |         -^^^^^^^^^^ method not found in `CrystalBuilder<LatticeSet, AtomsSet>`
   | |_________|
   |
   |
   = note: the method was found for
           - `CrystalBuilder<L, AtomsNotSet>`
